runtime/src/main/java/com/flipkart/flux/dao/StateMachinesDAOImpl.java

```diff
@@ -47,20 +50,20 @@ public StateMachine findById(Long id) {
 
     @Override
     @Transactional
-    public List<StateMachine> findByName(String stateMachineName) {
+    public Set<StateMachine> findByName(String stateMachineName) {
```

Comment:
Changed it to Set, as Hibernate is returning duplicate objects of same state machine, in case number of states are more than 1 (due to state machine and state tables join). Currently states fetch type is Eager, if we change it to Lazy, this issue can be avoided, but I don't want to change it, as if we try to access states outside transaction boundaries, we get "no session bound currently" exception (only in tests we are accessing states after session close right now).